okhttp/src/main/kotlin/okhttp3/internal/platform/ConscryptPlatform.kt

```diff
@@ -41,26 +43,18 @@ class ConscryptPlatform private constructor() : Platform() {
       SSLContext.getInstance("TLS", provider)
 
   override fun platformTrustManager(): X509TrustManager {
-    return Conscrypt.getDefaultX509TrustManager()
+    val trustManagers = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()).apply {
+      init(null as KeyStore?)
+    }.trustManagers!!
+    check(trustManagers.size == 1 && trustManagers[0] is X509TrustManager) {
+      "Unexpected default trust managers: ${trustManagers.contentToString()}"
+    }
+    val x509TrustManager = trustManagers[0] as X509TrustManager
+    Conscrypt.setHostnameVerifier(x509TrustManager) { _, _ -> true }
```

Comment:
@swankjesse Do we want to do this on Android generally also?  This could possibly help with the weird versioning story.